command/server_util.go

```diff
@@ -58,7 +58,7 @@ func adjustCoreForSealMigration(logger log.Logger, core *vault.Core, barrierSeal
 		// In this case our migration seal is set so we are using it
 		// (potentially) for unwrapping. Set it on core for that purpose then
 		// exit.
-		core.SetSealsForMigration(nil, nil, unwrapSeal)
+		core.SetSealsForMigration(unwrapSeal, barrierSeal, unwrapSeal)
```

Comment:
This is a change from my patch that fixes the migration from auto->auto of the same auto-type.  I don't really understand why it was the way it was, and I'm still concerned that although this solves one issue it might break something else.  @jefferai any thoughts?

Reply:
I honestly don't remember making this change. It could be a merge magic showing up here somehow. I'll remove this and do the testing once more.

Reply:
It's from my patch.

Reply:
And without it, auto->auto breaks if they're the same type.

Reply:
I see. I had read "change" in a different context. I tested this change and it is indeed required to make auto->auto work. I understand now that an open question from your PR is now an open question on this one.

Reply:
We/I didn't consider the auto->auto of same type case as a thing you'd want to do, it was assumed you'd only be migrating from one seal type to another. So the thought there is, the seal type in the block that's configured is the same seal type, therefore it _is_ the existing seal, so the other (disabled) seal is simply being used for unwrapping and we don't need to actually perform a migration.

If this change is necessary in order to perform a migration from _differently typed_ auto to auto then I don't really have any insight offhand, since that type check ought to be different then and you shouldn't run this code.

Reply:
No, it's only needed for auto->auto of the same type.  Thanks for clearing that up.

Reply:
Will sending this in now cause migration to run on every start up?

Reply:
The unwrapSeal arg to adjustCoresForSealMigration is only non-nil when there's a disabled seal section.  Given a nil unwrapSeal arg adjustCoreForSealMigration will return early: https://github.com/hashicorp/vault/blob/raft-seal-migration/command/server_util.go#L43

Reply:
I wonder what happens if user's don't update the config file after migration. Assuming that there is a disabled seal section all the time, there might still a chance that we run migration all the time. It doesn't happen with Raft though as per this PR. But its very easy to extend that logic to non-Raft migrations too. I'll have to test that out though (is still AFK, can test later).

Reply:
I've removed support for migration between same types for now. It was leading to complication in detecting if the migration has happened, during startup.